app/src/main/java/com/example/wrk/WorkoutsAdapter.java

```diff
@@ -122,6 +126,22 @@ public void bind(WorkoutPerformed workoutPerformed) throws ParseException, JSONE
 
                 tlWorkouts.addView(tbrow);      // link row to the tablelayout
             }
+            // double tap action to save a template
+            tlWorkouts.setOnTouchListener(new OnDoubleTapListener(mContext) {
+                @RequiresApi(api = Build.VERSION_CODES.N)
+                @Override
+                public void onDoubleTap(MotionEvent e) {
+                    // save template to users account if not already saved
+                    WorkoutTemplate tappedTemplate = workoutPerformed.getWorkout();
+                    if (!tappedTemplate.isSavedByCurrentUser()) {
+                        tappedTemplate.saveUserTemplate();
```

Comment:
Sorry not related to this PR, but wondering why unsaveTemplate() is called in saveUserTemplate()? 
Since you already checked if user has saved the template before calling saveUserTemplate(), it seems a bit redundant?

Reply:
> Sorry not related to this PR, but wondering why unsaveTemplate() is called in saveUserTemplate()? Since you already checked if user has saved the template before calling saveUserTemplate(), it seems a bit redundant?

Yeah I think you are right. Originally, I had used it without making the check and handled the check by unsaving it first to ensure that there were no duplicates of the people who have saved it. However, I feel that adding the check outside of the function call would likely make it more readable so that it isn't confusing as to why it is unsaved first within the save call. I will change this so that it performs the check outside of the function first so that the template is only saved if it passes the check. This should remove any redundancy. Thanks for catching that!